src/v/datalake/coordinator/coordinator.cc

```diff
@@ -146,4 +293,20 @@ coordinator::sync_get_last_added_offset(model::topic_partition tp) {
     co_return prt_state.pending_entries.back().last_offset;
 }
 
+void coordinator::notify_leadership(std::optional<model::node_id> leader_id) {
+    auto node_id = stm_->raft()->self().id();
+    bool is_leader = leader_id && *leader_id == stm_->raft()->self().id();
+    vlog(
+      datalake_log.debug,
+      "Coordinator leadership notification: is_leader: {}, leader_id: {}, self "
+      "node id: {}",
+      is_leader,
+      leader_id ? *leader_id : model::node_id{},
+      node_id);
+    if (is_leader) {
+        leader_cond_.signal();
```

Comment:
Yea Bharath suggested something similar. I'm going to keep it as is because I think it's a bit easier to work with the rough mental model of one fiber running at a time (e.g. nothing in the background quiescing from an old term). May revisit in the future though if this is shown to be problematic